### PR TITLE
fix for xss issue on description field for create order request

### PR DESCRIPTION
**Pull request by mpreddy928**
### Description

A DOM XSS exists in the description field of the [Smart Checkout Button](https://www.paypal.com/buttons/smart) feature, allowing an attacker to deploy a malicious smart button on a web page they control to exploit the DOM XSS in the victim's context and DOM

createOrder: function(data, actions) {

          return actions.order.create({

            purchase_units: [\{"description":"</scr" + "ipt><script src='https://www.paypal.com/merchantsignup/partner/mbcontent?callback=alert'></sc" + "ript><script>","amount":{"currency_code":"USD","value":1}}]

          });

        },

Escaping the Javascript context from the description field will result in the XSS. The hacker is providing a proof-of-concept on sandbox.paypal.com.

### Why are we making these changes? Include references to any related Jira tasks or GitHub Issues

See [DTXOUP-84](https://engineering.paypalcorp.com/jira/browse/DTXOUP-84)

### CODE REVIEW

File: src/props/createOrder.js [hunk 1]
```diff
@@ -57,6 +57,9 @@ export function buildOrderActions({ facilitatorAccessToken, intent, currency, me
         order = { ...order, intent: intent.toUpperCase() };
     
         order.purchase_units = order.purchase_units.map(unit => {
+            if (unit.description) {
+                unit.description = escape(unit.description);
```
[the remark this answers is not in the document]
Comment (mpreddy928) [reply src/props/createOrder.js:61]: @westeezy how about using lodash _.escape ?

[the remark this answers is not in the document]
Comment (mpreddy928) [reply src/props/createOrder.js:61]: or this is the better example considering the example you gave https://stackoverflow.com/a/48226843

**Comment by bluepnume** [comment]
This is not a good way to fix this issue. Attackers can still call the api independently of this client side javascript code. We need to escape at the point in the app where this is being rendered -- not here.

File: src/props/createOrder.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (westeezy) [reply src/props/createOrder.js:61]: We have a similar function to lodash's escape in belter here: [htmlEncode](https://github.com/krakenjs/belter/blob/main/src/dom.js#L251)

**Comment by westeezy** [comment]
I agree with @bluepnume. This could also potentially cause issues down stream because we are mangling the description which should be fine as plaintext. We just always have to be careful when we render user inputted data.

File: src/props/createOrder.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (mpreddy928) [reply src/props/createOrder.js:61]: thanks

**Comment by westeezy** [comment]
Closing issue to be fixed downstream when this value is rendered.

**Pull request closed without merging**